File: iosApp/iosAppTests/More/MoreStateRobot.swift
import ComposeApp

final class MoreStateRobot: BaseStateRobot<MoreUiState, MoreEvent> {

    override func defaultState() -> MoreUiState {
        MoreUiState(
            userProfile: UserProfile(name: "Test User", tier: "Gold", points: "2,450 pts", avatarUrl: ""),
            menuItems: [
                MoreMenuItem(id: "1", icon: "⚙️", title: "Settings"),
                MoreMenuItem(id: "2", icon: "❓", title: "Help"),
            ],
            eventSink: createEventSink()
        )
    }

    func stateWithNoProfile() -> MoreUiState {
        MoreUiState(
            userProfile: nil,
            menuItems: [
                MoreMenuItem(id: "1", icon: "⚙️", title: "Settings"),
                MoreMenuItem(id: "2", icon: "❓", title: "Help"),
            ],
            eventSink: createEventSink()
        )
    }

    func stateWithEmptyMenu() -> MoreUiState {
        MoreUiState(
            userProfile: UserProfile(name: "Test User", tier: "Gold", points: "2,450 pts", avatarUrl: ""),
            menuItems: [],
            eventSink: createEventSink()
        )
    }
}

File: features/more/impl/presentation/src/androidDeviceTest/kotlin/com/mockdonalds/app/features/more/presentation/MoreStateRobot.kt
package com.mockdonalds.app.features.more.presentation

import com.mockdonalds.app.core.test.StateRobot
import com.mockdonalds.app.features.more.api.domain.MoreMenuItem
import com.mockdonalds.app.features.more.api.domain.UserProfile

class MoreStateRobot : StateRobot<MoreUiState, MoreEvent>() {

    override fun defaultState() = MoreUiState(
        userProfile = UserProfile(
            name = "TestUser",
            tier = "Gold",
            points = "1,250 PTS",
            avatarUrl = "",
        ),
        menuItems = listOf(
            MoreMenuItem(id = "1", icon = "📍", title = "Find a Restaurant"),
            MoreMenuItem(id = "2", icon = "⚙️", title = "Settings"),
        ),
        eventSink = createEventSink(),
    )

    fun stateWithNoProfile() = defaultState().copy(
        userProfile = null,
        eventSink = createEventSink(),
    )

    fun stateWithEmptyMenu() = defaultState().copy(
        menuItems = emptyList(),
        eventSink = createEventSink(),
    )
}
